### ISSUE TITLE
Fix and reenable SimpleContextPropagationTest.testArcMEContextPropagationDisabled

### ISSUE THREAD

**Issue by gsmet** [quarkusio/quarkus#28129]
It has been disabled here because it was flaky: https://github.com/quarkusio/quarkus/pull/28127 .

/cc @mkouba @FroMage

**Comment by manovotn** [comment]
@gsmet got link to some PRs that failed this?
I have been trying to reproduce this with a loop execution locally but so far not a single failure.

BTW this is the same test that we tried to fix in https://github.com/quarkusio/quarkus/pull/27284
Although back then I think I saw at least some failures locally...

**Comment by gsmet** [comment]
Yes, here for instance: https://github.com/quarkusio/quarkus/pull/28121

**Comment by geoand** [comment]
Yes, it makes total sense. The combination of using reactive types with RESTEasy Classic is just too problematic (and is one of the reasons why we did RR in the first place).

**Comment by gsmet** [comment]
BTW, I'm wondering if we could get rid of `io.reactivex.Single` now? @cescoffier WDYT?

**Comment by cescoffier** [comment]
@gsmet yes, probably time to remove the RX java 2 support

**Comment by geoand** [comment]
Definitely +1 for that
